Add new Style Guide template to incubator (#4)
thegooddocsproject/incubator: style-guide/about-style-guide.md

guide that are most relevant to your project or point out any exceptions to the
default guide.

As your project grows and matures, your style guide can grow along with it. For
example, debates can (and will) arise about how to refer to a specific term in
the documentation, such as whether to capitalize or hyphenate a term that is
unique to the project. Once resolved, you can capture decisions in the style
guide. Then, if that debate arises again, you can simply refer contributors to
your style guide rather than rehashing the same debate every time a new
contributor joins the project.

However, keep in mind that some teams consistently struggle to maintain a style
guide over time. The guiding principle here is that a small amount of work up
front can save a lot more work later. It’s much easier to set your standards in
a style guide early in your project’s life cycle than it is to retroactively
clean up your documentation later when the project is more mature. With small
but regular updates, your style guide can evolve alongside your project. It will
help your contributors get up to speed quickly and start writing high quality
documentation.

> To help your project scale, consider setting a revolving meeting or agenda item

Reviewer: I have a mild preference for "periodic meeting" over "revolving meeting".
Feel free to ignore.

Author: I agree with you that I like "periodic" better than "revolving," but I've also just thought that "recurring" might be good. I've changed it to "recurring" for now, but I'm open to using "periodic" as well.

to review the style guide and address style guide issues on a periodic basis
(such as once a month). Perhaps you could make someone on the team the project
owner for the style guide to ensure someone has the responsibility to update the
guide after style guide reviews.

> 

Reviewer: I think you should add a section titled "Target audience"
In it you describe that your audience for a style guide are the people who write your documentation. You will know them best. They typically will include software engineers, product mangers, tech writers if you are lucky enough to have them on the team, ...

Author: Good suggestion. I've added a section for "Intended audience and scope" to both the template and the template description.

you intend to use this style guide in your project. Feel free to adapt this
content. If desired, you can briefly elaborate on the goals or mission of your
project’s documentation.

### About the "Our preferred style guide" section

Maintenance of a style guide is a lot of work. Try to do as little of it as
possible by deferring to a more comprehensive default guide, such as the Google
Developer's Style Guide or the Microsoft Manual of Style. In this section,
you should indicate which guide is your default guide and reference the source
document and its version. Document any exceptions where your style guide differs
from your default guide.

Also, if your project already has existing brand guidelines, you can link to
them or include them here.

## About the "Glossary of preferred terms" section

Every project has its own unique terms and buzzwords and this glossary can help
you record your project’s preferred terms. This word list provides a list of
specific terms that are either common word choice problems or which are unique
to your project. As a word of caution, try to minimize jargon and buzzwords.
Also, avoid creating too many project-specific terms. Where possible, use the

> terms that are standard for your industry to avoid confusion.

Reviewer: Suggest add:
Ideally you can defer to, or inherit term definitions from a more authoritative glossary source.

Author: Great addition! I'v adopted it.

This section includes a table to highlight or call out any terms that are unique
or which conflict with your default style guide. For example, the Google
Developer's Guide uses "open source" but our project prefers "open-source" when
used as a modifier, as in "open-source software."

## About the "Topic types and templates" section

Depending on the type of project you’re creating, some of the Good Docs
templates will be more relevant than others. In this section, you can link to
the templates you have adopted for your project. You can also explain when
contributors should use these templates.

## About the "General writing tips" section

This section is optional. You can link to your favorite general writing tips or
add some of your own general guidelines.

## About the "Accessible writing" section

In this section, you can link to your preferred style guides about accessible
writing or add some of your own general guidelines.

## About the "Inclusive and bias-free writing" section

In this section, you can link to your favorite resources about inclusive and
bias-free writing or add some of your own general guidelines.

## About the "Using linters" section

> This section is optional. [Linters](https://www.writethedocs.org/guide/tools/testing/#style-guide-checking-and-linting)

Reviewer: No need to change this now, but long term I think that we should link to more authoritative docs than provided by www.writethdocs.org. This link has the potential to be changed and cause link rot.

Author: That's a great point and I'll make a note of that.

post-processing test of the content when the document is pushed into a revision
management system as part of an automated continuous integration (CI) cycle.

If you want to take your style guide to the next level, you could adopt a linter
to enforce your style guide and empower your content creators to make better
documentation with less cognitive overhead.

For resources and examples of linters, see:

- [The `linter-google-styleguide` package for the Atom text editor](https://atom.io/packages/linter-google-styleguide)
- A tool like [Vale](https://www.writethedocs.org/guide/tools/testing/#vale),
  which can automate [several popular styles](https://github.com/errata-ai/styles)

## About the "How the style guide is updated" section

Indicate here how frequently your style guide is reviewed, who owns the style
guide, and how contributors can provide feedback on your style guide. For
example, you could provide the contact information for the group or individual
who owns the style guide. You could also indicate when the style guide is
reviewed in project meetings.

If this section is small enough, you can combine it with the following sections.

## About the "Revision history" section

> 

Reviewer: I feel all authoritative documents, such as a style guide, should be allocated a version number. Ideally, it would have a "Last updated" field too, which could be tied just to a month, instead of just a day.
Reason: People should be able to say "My docs have been reviewed against version X.X of the <project> style guide and comply."
Business decisions can the be made about whether the docs should be reviewed again based on the style guide changing.
Typically, this info will be displayed at the top of the doc. (So consider applying actions from this comment to the top).

Author: I agree with your suggestion. I've added this information to the top of the template document.

> # The style guide

Reviewer 2: This starts with sentence case, then switches to title case. Since TGD's own style guide suggests Google's style guide, I think sentence case is probably the right choice here

Author: Yeah, I debated that. I'll go through and make these changes.

The style guide template itself occasionally includes placeholder text that
is intended for you to replace with your own text. This text is indicated in
{curly brackets}.

## When do I need a style guide?

A style guide provides project contributors with general guidelines for writing
project documentation. The overall goal of a style guide is to ensure quality
and consistency throughout the project’s documentation, which is especially
important if different authors are contributing to the documentation over time.

A style guide can provide early guidance to new contributors as they make
decisions related to your project’s documentation. Contributors often have to

> make subjective judgment calls about the documentation, such as:

Reviewer 3: The phrase "judgement calls" might be slang. I suggest "decisions" instead.

Author: I like that suggestion! I've adopted it.

- Which style guide your project will defer to as its default style guide
- How to refer to terms that are unique or specific to that project
- How to format certain types of content
- What tone, attitude, or level of formality you recommend using in the project
  (usually conveyed through word choice)

> - Any specific rules you want to highlight or call out to documentation

Reviewer 3: Maybe remove "or call out"? "Highlight" should be enough.

Author: I also like this suggestion and I've adopted it. 👍

## Content of your style guide

### About the "Introduction" section

In this section, you'll briefly introduce the purpose of the style guide and how
you intend to use this style guide in your project. Feel free to adapt this
content. If desired, you can briefly elaborate on the goals or mission of your
project’s documentation.

### About the "Our Preferred Style Guide" section

Maintenance of a style guide is a lot of work. Try to do as little of it as
possible by deferring to a more comprehensive default guide, such as the Google
Developer's Style Guide or the Microsoft Manual of Style. In this section,
you should indicate which guide is your default guide and reference the source
document and its version. Document any exceptions where your style guide differs
from your default guide.

Also, if your project already has existing brand guidelines, you can link to
them or include them here.

## About the "Glossary of Preferred Terms" section

Every project has its own unique terms and buzzwords and this glossary can help
you record your project’s preferred terms. This word list provides a list of
specific terms that are either common word choice problems or which are unique

> to our project. As a word of caution, try to minimize jargon and buzzwords.

Reviewer 3: Is this "our" supposed to be "your" ?

Author: Another good catch. Thank you!

Also, avoid creating too many project-specific terms. Where possible, use the
terms that are standard for your industry to avoid confusion.

This section includes a table to highlight or call out any terms that are unique
or which conflict with your default style guide. For example, the Google
Developer's Guide uses "open source" but our project prefers "open-source" when
used as a modifier, as in "open-source software."

## About the "Topic Types and Templates" section

Depending on the type of project you’re creating, some of the Good Docs
templates will be more relevant than others. In this section, you can link to
the templates you have adopted for your project. You can also explain when
contributors should use these templates.

## About the "General Writing Tips" section

> This seciton is optional. You can link to your favorite general writing tips or

Reviewer 3: seciton typo argh!

Author: Thanks for the catch! I've fixed it.

writing or add some of your own general guidelines.

## About the "Inclusive and Bias-Free Writing" section

In this section, you can link to your favorite resources about inclusive and
bias-free writing or add some of your own general guidelines.

## About the "Using Linters" section

This section is optional. [Linters](https://www.writethedocs.org/guide/tools/testing/#style-guide-checking-and-linting)
are tools that analyze text content to flag common mistakes such as stylistic
errors and suspicious constructs. An example of a linter would be a tool like
Grammarly that processes your text and provides suggestions as you write to
improve the spelling and grammatical structure of the text.

Linters can be used during the editorial process within an editor and as a
post-processing test of the content when the document is pushed into a revision
management system as part of an automated continuous integration (CI) cycle.

If you want to take your style guide to the next level, you could adopt a linter
to enforce your style guide and empower your content creators to make better
documentation with less cognitive overhead.

For resources and examples of linters, see:

> - [The Atom editor’s Google Linter app](https://atom.io/packages/linter-google-styleguide)

Reviewer 2: I found "app" confusing here

```suggestion
- [The `linter-google-styleguide` package for the Atom text editor](https://atom.io/packages/linter-google-styleguide)
```

- A tool like [Vale](https://www.writethedocs.org/guide/tools/testing/#vale)
  for a more automated approach, including styles such as the

>   [Microsoft Writing Style Guide](https://github.com/errata-ai/styles)

Reviewer 2: I expected this link to take me directly to the MS style for Vale. I'd consider reworking this. Perhaps something like:

> A tool like [Vale](https://www.writethedocs.org/guide/tools/testing/#vale), which can automate [several popular styles](https://github.com/errata-ai/styles)

Author: That's good feedback! I liked that suggestion and I've adopted it.

# The style guide

> The style guide template itself occasionally includes placeholder text that

Reviewer 4: All of these are hard line breaks, is that intentional? Perhaps go with a one sentence per line style instead if you want a hard wrap?

Author: This is just the standard at my workplace, but I'm open to suggestions.

Reviewer 4: Huh, that's interesting. I don't think it's worth changing, but it might be worth discussing if we want to have a project standard for this.

Reviewer 2: If we care about source style for Markdown, then we may want to adopt a tool like [Prettier](https://prettier.io/). It would push the responsibility of choosing a style onto a tool (so we don't have to debate it) and makes keeping to that style a snap (because the tool does it for you).

Opened an issue for this idea: https://github.com/thegooddocsproject/templates/issues/160

Author: Thanks for opening that @ddbeck . @Loquacity , I'd definitely be open to changing this once we decide on the line break standard. Line break debates seem to be a constant point of division at any workplace I go to, but I don't personally have a dog in the fight, so I'll conform with whatever we decide.

is intended for you to replace with your own text. This text is indicated in
{curly brackets}.

## When do I need a style guide?

A style guide provides project contributors with general guidelines for writing
project documentation. The overall goal of a style guide is to ensure quality

> and consistency throughout the project’s documentation, which is especially

Reviewer 4: It looks like you have smart apostrophes in some parts, and straight in others. I would recommend a find and replace to make them all straight, just so you don't run into problems with displaying the smart ones later on.

Author: A very good catch. If we have the bandwidth, it might be nice to install some kind of pre-commit check for this. I've opened an issue for consideration: https://github.com/thegooddocsproject/templates/issues/159

guide. Then, if that debate arises again, you can simply refer contributors to
your style guide rather than rehashing the same debate every time a new
contributor joins the project.

However, keep in mind that some teams consistently struggle to maintain a style
guide over time. The guiding principle here is that a small amount of work up
front can save a lot more work later. It’s much easier to set your standards in
a style guide early in your project’s life cycle than it is to retroactively
clean up your documentation later when the project is more mature. With small
but regular updates, your style guide can evolve alongside your project. It will
help your contributors get up to speed quickly and start writing high quality
documentation.

To help your project scale, consider setting a recurring meeting or agenda item
to review the style guide and address style guide issues on a periodic basis
(such as once a month). Perhaps you could make someone on the team the project
owner for the style guide to ensure someone has the responsibility to update the
guide after style guide reviews.

## Content of your style guide

### About the "Introduction" section

> In this section, you'll briefly introduce the purpose of the style guide and how

Reviewer 4: ```suggestion
In this section, briefly introduce the purpose of the style guide and how
```

documentation.

### About the "Our preferred style guide" section

Maintenance of a style guide is a lot of work. Try to do as little of it as
possible by deferring to a more comprehensive default guide, such as the Google
Developer's Style Guide or the Microsoft Manual of Style. In this section,
you should indicate which guide is your default guide and reference the source
document and its version. Document any exceptions where your style guide differs
from your default guide.

Also, if your project already has existing brand guidelines, you can link to
them or include them here.

## About the "Glossary of preferred terms" section

Every project has its own unique terms and buzzwords and this glossary can help
you record your project’s preferred terms. This word list provides a list of
specific terms that are either common word choice problems or which are unique
to your project. As a word of caution, try to minimize jargon and buzzwords.
Also, avoid creating too many project-specific terms. Where possible, use the
terms that are standard for your industry to avoid confusion. Ideally, you can
inherit or defer to term definitions from a more authoritative glossary source.

> This section includes a table to highlight or call out any terms that are unique

Reviewer 4: ```suggestion
This section includes a table to highlight terms that are unique to your project,
```

or which conflict with your default style guide. For example, the Google
Developer's Guide uses "open source" but our project prefers "open-source" when

> used as a modifier, as in "open-source software."

Reviewer 4: Do you mean "adjective" here?

Author: In my mind, they're co-equal. I'll change to adjective if you'd like since I don't have a strong preference. Adopted.

## About the "Topic types and templates" section

Depending on the type of project you’re creating, some of the Good Docs
templates will be more relevant than others. In this section, you can link to
the templates you have adopted for your project. You can also explain when
contributors should use these templates.

> 

Reviewer 4: The heading mentions topic types but you haven't explained that.

Author: I've added this sentence: "The Good Docs project provides templates for various types of topics that you can adopt in your project." Feel free to suggest any improvements on that!

## About the "General writing tips" section

This section is optional. You can link to your favorite general writing tips or
add some of your own general guidelines.

## About the "Accessible writing" section

In this section, you can link to your preferred style guides about accessible
writing or add some of your own general guidelines.

## About the "Inclusive and bias-free writing" section

In this section, you can link to your favorite resources about inclusive and
bias-free writing or add some of your own general guidelines.

## About the "Using linters" section

This section is optional. [Linters](https://www.writethedocs.org/guide/tools/testing/#style-guide-checking-and-linting)
are tools that analyze text content to flag common mistakes such as stylistic
errors and suspicious constructs. An example of a linter would be a tool like
Grammarly that processes your text and provides suggestions as you write to
improve the spelling and grammatical structure of the text.

> Linters can be used during the editorial process within an editor and as a

Reviewer 4: ```suggestion
Linters can be used during the editorial process within a text editor and as a
```
To distinguish the tool from the human ;)

post-processing test of the content when the document is pushed into a revision

> management system as part of an automated continuous integration (CI) cycle.

Reviewer 4: This is a very long sentence!

Author: My edit:
Linters can be used at two different points when creating documentation:
- **During the editorial process** - Typically used within a text editor, a
  linter can check for style consistency while composing documentation.
- **As a post-processing test** - When the content is pushed into a revision
  management system as part of an automated continuous integration (CI) cycle,
  a linter can evaluate text for style consistency.
- **As a post-processing test** - When the content is pushed into a revision
  management system as part of an automated continuous integration (CI) cycle,
  a linter can evaluate text for style consistency.

Reviewer 4: I think you need to delete the third bullet there, but otherwise that looks fine to me :+1:

Author: That's odd that it got in there twice in my copy. It's correct in the version I pushed at least. Should be good now!

If you want to take your style guide to the next level, you could adopt a linter
to enforce your style guide and empower your content creators to make better
documentation with less cognitive overhead.

For resources and examples of linters, see:

- [The `linter-google-styleguide` package for the Atom text editor](https://atom.io/packages/linter-google-styleguide)
- A tool like [Vale](https://www.writethedocs.org/guide/tools/testing/#vale),

>   which can automate [several popular styles](https://github.com/errata-ai/styles)

Reviewer 4: It might also be worth noting something like Alex, which lints for gender and race neutral language (and similar), to make it clear that linters are for more than just spelling and grammar.

Author: Good suggestion! I've added a link and a description.
